omegaconf/dictconfig.py

```diff
@@ -130,30 +131,13 @@ def __deepcopy__(self, memo: Dict[int, Any] = {}) -> "DictConfig":
             content_copy = src_content
 
         res.__dict__["_content"] = content_copy
-
+        # parent is retained, but not copied
+        res.__dict__["_parent"] = self.__dict__["_parent"]
         return res
 
     def __copy__(self) -> "DictConfig":
```

Comment:
Move `__copy__` to the `BaseContainer` class? Or do we need it here for typing purpose?